src/promise-cache.ts

```diff
@@ -69,19 +72,30 @@ export class PromiseCache<T> {
         return this.stats.export(this.cache.size);
     }
 
-    private cleanUp() {
+    cleanUp() { //  needs to be non-private so that we can call it from tests.  need to call it from tests to achieve full branch coverage.
         const now = Date.now();
 
         // workaround as for(const it of this.cache.entries()) does not work
         Array.from(this.cache.entries()).forEach((it) => {
             const [key, entry] = it;
-            if ((entry.lastAccess + (this.conf.ttl as number)) < now) {
-                try {
-                    this.conf.onRemove(key, entry.value);
-                } catch (error) {
-                    // nothing we can do
+            if (this.conf.ttl !== "FOREVER") {
```

Comment:
actually - now that there is only one ttl config variable, I could put this back to the way it was and not have this !== "FOREVER" check -- and thus make .cleanUp private again.  Let me know if you'd prefer that.